addons/account/models/account_move.py

```diff
@@ -4123,6 +4123,16 @@ def button_draft(self):
         if any(move.state not in ('cancel', 'posted') for move in self):
             raise UserError(_("Only posted/cancelled journal entries can be reset to draft."))
 
+        self._checks_action_draft()
+
+        for move in self:
+            # We remove all the analytics entries for this journal
+            move.mapped('line_ids.analytic_line_ids').unlink()
+
+        self.mapped('line_ids').remove_move_reconcile()
+        self.write({'state': 'draft', 'is_move_sent': False})
+
+    def _checks_action_draft(self):
```

Comment:
I would maybe name the method `_check_draftable` instead, i would find it easier to get the point of the method.